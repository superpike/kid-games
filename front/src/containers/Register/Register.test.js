import React from 'react';
import { render } from '@testing-library/react';
import { Register } from './index';
import '@testing-library/jest-dom/extend-expect';

describe('Testing Register container', () => {
  const fakeRegister = jest.fn(data => data);
  const fakeCancel = jest.fn(data => data);

  test('should match to snapshot', () => {
    const { container } = render(<Register />);
    expect(container).toMatchSnapshot();
  });

  describe('Buttons clicks', () => {
    test('should register', () => {
      const { getByTestId } = render(
        <Register register={fakeRegister} cancel={fakeCancel} />
      );
      const button = getByTestId('register');
      button.click();
      expect(fakeRegister).toHaveBeenCalledTimes(1);
    });
    test('should cancel registration', () => {
      const { getByTestId } = render(
        <Register register={fakeRegister} cancel={fakeCancel} />
      );
      const button = getByTestId('cancel');
      button.click();
      expect(fakeCancel).toHaveBeenCalledTimes(1);
    });
  });
});
