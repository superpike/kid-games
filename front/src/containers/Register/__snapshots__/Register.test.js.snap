// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Register container should match to snapshot 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="wrapper"
    >
      <label
        class="label"
        for="username"
      >
        username
      </label>
      <input
        class="InputElement"
        name="username"
        type="text"
        value=""
      />
    </div>
    <div
      class="wrapper"
    >
      <label
        class="label"
        for="email"
      >
        email
      </label>
      <input
        class="InputElement"
        name="email"
        type="text"
        value=""
      />
    </div>
    <div
      class="wrapper"
    >
      <label
        class="label"
        for="password"
      >
        password
      </label>
      <input
        class="InputElement"
        name="password"
        type="password"
        value=""
      />
    </div>
    <button
      class="Main Button"
      data-testid="register"
      type="button"
    >
      register
    </button>
    <button
      class="Danger Button"
      data-testid="cancel"
      type="button"
    >
      cancel
    </button>
  </div>
</div>
`;
